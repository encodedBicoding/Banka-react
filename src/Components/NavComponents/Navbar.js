import React, { useEffect, useState } from 'react';
import BigNav from './BigNav.js';
import SmallNav from './SmallNav.js';
import './NavStyle.css';

const Navbar = props => {
  let [scrollPosition, setScrollPosition] = useState(0);
  useEffect(() => {
    window.onscroll = () => {
      setScrollPosition(document.documentElement.scrollTop);
    };
  }, [scrollPosition]);
  return (
    <div className="main-container">
      {scrollPosition > 20 ? (
        <SmallNav
          home={props.home}
          about={props.about}
          contact={props.contact}
          signup={props.signup}
          login={props.login}
        />
      ) : (
        <BigNav
          home={props.home}
          about={props.about}
          contact={props.contact}
          signup={props.signup}
          login={props.login}
        />
      )}
    </div>
  );
};

export default Navbar;
