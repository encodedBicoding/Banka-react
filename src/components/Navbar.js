import React from "react";
import { Link } from "react-router-dom";

const Navbar = props => {
  return (
    <React.Fragment>
      <div className="nav_logo">
        <span className="nav_title">B</span>
        <span className="nav_title">A</span>
        <span className="nav_title">N</span>
        <span className="nav_title">K</span>
        <span className="nav_title nav_last_child">A</span>
        <p className="nav_slogan">...first online mobile banking service</p>
      </div>
      <nav className="home_nav">
        <ul className="home_nav_header">
          <li className="home_nav_list">
            <Link to="/" id="nav_link">
              {props.home}
            </Link>
          </li>
          <li className="home_nav_list">
            <Link to="#about" id="nav_link">
              {props.about}
            </Link>
          </li>
          <li className="home_nav_list">
            <Link to="#contact" id="nav_link">
              {props.contact}
            </Link>
          </li>
          <li className="home_nav_list">{props.signup}</li>
          <li className="home_nav_list">{props.login}</li>
        </ul>
      </nav>
    </React.Fragment>
  );
};

export { Navbar };
