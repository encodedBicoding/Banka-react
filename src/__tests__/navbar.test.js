import React from 'react';
import Navbar from '../Components/NavComponents/Navbar';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Handle Navbar Component test', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });
});
