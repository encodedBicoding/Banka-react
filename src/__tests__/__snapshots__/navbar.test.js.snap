// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handle Navbar Component test Should render correctly 1`] = `
<div
  className="main-container"
>
  <Connect(BigNav) />
</div>
`;
